velox/functions/prestosql/aggregates/SetAggregates.cpp

```diff
@@ -274,9 +287,32 @@ class SetAggAggregate : public SetBaseAggregate<T> {
 
     auto tracker = Base::trackRowSize(group);
     rows.applyToSelected([&](vector_size_t i) {
+      if (throwOnNestedNulls_) {
+        checkNulls(Base::decoded_, i);
+      }
+
       accumulator->addValue(Base::decoded_, i, Base::allocator_);
     });
   }
+
+ private:
+  bool checkNulls(const DecodedVector& decoded, vector_size_t index) const {
+    if (decoded.isNullAt(index)) {
+      return true;
+    }
+
+    if (throwOnNestedNulls_) {
+      VELOX_USER_CHECK(
+          !decoded.base()->containsNullAt(index),
```

Comment:
This is incorrect. We should use decoded.index(index)

```
!decoded.base()->containsNullAt(decoded.index(index)),
```

Same problem exists in min/max aggregate function.

Reply:
Got it, must use the base index! I've fixed it along with the min/max aggregate function.